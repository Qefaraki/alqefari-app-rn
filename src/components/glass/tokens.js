// Design tokens for the liquid-glass aesthetic
// Centralized so we can tune the look consistently across components

export const glassTokens = {
  radii: {
    xs: 12,
    sm: 16,
    md: 20,
    lg: 28,
    xl: 36,
  },
  blurIntensity: {
    subtle: 28,
    medium: 40,
    strong: 64,
  },
  // Base surface overlay colors for a bright UI
  surfaceOverlay: {
    // Vertical overlay from slightly brighter top to slightly dimmer bottom
    gradient: ['rgba(255,255,255,0.55)', 'rgba(255,255,255,0.28)'],
  },
  // Hairline borders that feel like condensed water on the glass
  borders: {
    inner: 'rgba(255,255,255,0.55)',
    outer: 'rgba(0,0,0,0.04)',
  },
  // Soft ambient elevation â€“ avoid harsh shadows
  elevation: {
    ios: {
      shadowColor: '#000',
      shadowOpacity: 0.10,
      shadowRadius: 14,
      shadowOffset: { width: 0, height: 8 },
    },
    android: {
      elevation: 3,
    },
  },
  highlight: {
    // Specular highlight sweeping from top-left
    sheen: ['rgba(255,255,255,0.7)', 'rgba(255,255,255,0.0)'],
  },
};

export default glassTokens;


