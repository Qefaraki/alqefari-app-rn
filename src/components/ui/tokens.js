// Design tokens for neo-native (non-glass) UI
// Neutral surfaces, soft elevation, crisp dividers, Arabic-first
// Aligned with iOS Human Interface Guidelines

const tokens = {
  colors: {
    bg: '#F2F2F7',
    surface: '#FFFFFF',
    text: '#111827',
    textMuted: '#6B7280',
    divider: 'rgba(0,0,0,0.08)',
    outline: 'rgba(0,0,0,0.12)',
    accent: '#007AFF',
    success: '#34C759',
    danger: '#FF3B30',
    najdi: {
      background: '#F9F7F3', // Al-Jass White
      container: '#D1BBA3', // Camel Hair Beige
      text: '#242121', // Sadu Night
      textMuted: '#736372',
      primary: '#A13333', // Najdi Crimson
      secondary: '#D58C4A', // Desert Ochre
      focus: '#957EB5',
    },
    diff: {
      added: '#34C759',       // iOS green for new values
      addedBg: '#34C75915',   // 8% opacity
      removed: '#FF3B30',     // iOS red for old values
      removedBg: '#FF3B3015', // 8% opacity
      modified: '#007AFF',    // iOS blue for changes
      modifiedBg: '#007AFF15', // 8% opacity
    },
  },
  // iOS-standard typography scale
  typography: {
    largeTitle: { fontSize: 34, fontWeight: '700', lineHeight: 41 },
    title1: { fontSize: 28, fontWeight: '700', lineHeight: 34 },
    title2: { fontSize: 22, fontWeight: '700', lineHeight: 28 },
    title3: { fontSize: 20, fontWeight: '600', lineHeight: 25 },
    headline: { fontSize: 17, fontWeight: '600', lineHeight: 22 },
    body: { fontSize: 17, fontWeight: '400', lineHeight: 22 },
    callout: { fontSize: 16, fontWeight: '400', lineHeight: 21 },
    subheadline: { fontSize: 15, fontWeight: '400', lineHeight: 20 },
    footnote: { fontSize: 13, fontWeight: '400', lineHeight: 18 },
    caption1: { fontSize: 12, fontWeight: '400', lineHeight: 16 },
    caption2: { fontSize: 11, fontWeight: '400', lineHeight: 13 },
  },
  radii: {
    sm: 10,
    md: 12,
    lg: 16,
    xl: 20,
    full: 9999,  // Full pill shape for chips
  },
  // iOS-standard spacing scale (8px grid)
  spacing: {
    xxs: 4,
    xs: 8,
    sm: 12,
    md: 16,
    lg: 20,
    xl: 24,
    xxl: 32,
  },
  // iOS minimum touch target
  touchTarget: {
    minimum: 44,
  },
  shadow: {
    ios: {
      shadowColor: '#000',
      shadowOpacity: 0.07,
      shadowRadius: 12,
      shadowOffset: { width: 0, height: 8 },
    },
    android: {
      elevation: 3,
    },
  },
};

export default tokens;
