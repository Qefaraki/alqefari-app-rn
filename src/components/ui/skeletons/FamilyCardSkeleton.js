import React from 'react';
import { View, StyleSheet } from 'react-native';
import Shimmer from './Shimmer';
import tokens from '../tokens';

/**
 * FamilyCardSkeleton - Loading skeleton for FamilyCard
 * Matches InfoCard wrapper with horizontal family tiles
 */
const FamilyCardSkeleton = ({ tileCount = 4 }) => {
  return (
    <View
      style={styles.container}
      accessible={true}
      accessibilityLabel="جاري تحميل بطاقة العائلة"
      accessibilityRole="progressbar"
    >
      {/* Title */}
      <Shimmer
        width={80}
        height={22}
        borderRadius={tokens.radii.sm}
        style={styles.title}
      />

      {/* Divider */}
      <View style={styles.divider} />

      {/* Horizontal row of family tiles */}
      <View style={styles.familyRow}>
        {Array.from({ length: tileCount }).map((_, index) => (
          <View key={`tile-skeleton-${index}`} style={styles.tile}>
            {/* Avatar circle */}
            <Shimmer
              width={40}
              height={40}
              borderRadius={20}
              style={styles.avatar}
            />
            {/* Name */}
            <Shimmer
              width={70}
              height={16}
              borderRadius={tokens.radii.sm}
              style={styles.tileName}
            />
            {/* Label (optional, alternating) */}
            {index % 3 === 0 && (
              <Shimmer
                width={50}
                height={14}
                borderRadius={tokens.radii.sm}
              />
            )}
          </View>
        ))}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: tokens.colors.najdi.background, // Al-Jass White
    borderRadius: tokens.radii.md, // 12px
    paddingHorizontal: tokens.spacing.md, // 16px
    paddingTop: tokens.spacing.md, // 16px
    paddingBottom: tokens.spacing.lg, // 20px
    marginBottom: tokens.spacing.md, // 16px

    // iOS-style elevation
    ...tokens.shadow.ios,

    // Subtle Najdi border
    borderWidth: 1,
    borderColor: tokens.colors.najdi.container + '40', // Camel Hair 40%
  },
  title: {
    marginBottom: tokens.spacing.sm, // 12px
  },
  divider: {
    height: 1,
    backgroundColor: tokens.colors.najdi.container + '20',
    marginVertical: tokens.spacing.sm, // 12px
  },
  familyRow: {
    flexDirection: 'row',
    paddingVertical: 4,
    paddingHorizontal: 2,
    gap: 12,
  },
  tile: {
    alignItems: 'center',
    gap: 8,
    width: 80,
  },
  avatar: {
    marginBottom: 4,
  },
  tileName: {
    marginBottom: 4,
  },
});

export default FamilyCardSkeleton;
