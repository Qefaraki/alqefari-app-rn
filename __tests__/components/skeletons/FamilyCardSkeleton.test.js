/**
 * FamilyCardSkeleton Component Test Suite
 *
 * Tests the family card loading skeleton with tile layout
 */

import React from 'react';
import { render } from '@testing-library/react-native';
import FamilyCardSkeleton from '../../../src/components/ui/skeletons/FamilyCardSkeleton';
import tokens from '../../../src/components/ui/tokens';

// Mock dependencies
jest.mock('../../../src/components/ui/skeletons/Shimmer', () => 'Shimmer');

describe('FamilyCardSkeleton Component', () => {
  describe('Rendering', () => {
    it('renders without crashing', () => {
      const { toJSON } = render(<FamilyCardSkeleton />);
      expect(toJSON()).toBeTruthy();
    });

    it('renders with default tileCount (4)', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton />);

      // Find all tile containers
      const tiles = UNSAFE_root.findAllByProps({ style: expect.objectContaining({ width: 80 }) });

      expect(tiles).toHaveLength(4);
    });

    it('renders with custom tileCount', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton tileCount={6} />);

      const tiles = UNSAFE_root.findAllByProps({ style: expect.objectContaining({ width: 80 }) });

      expect(tiles).toHaveLength(6);
    });

    it('renders with tileCount=1', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton tileCount={1} />);

      const tiles = UNSAFE_root.findAllByProps({ style: expect.objectContaining({ width: 80 }) });

      expect(tiles).toHaveLength(1);
    });

    it('renders with tileCount=0 (edge case)', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton tileCount={0} />);

      const tiles = UNSAFE_root.findAllByProps({ style: expect.objectContaining({ width: 80 }) });

      expect(tiles).toHaveLength(0);
    });
  });

  describe('Card Structure', () => {
    it('renders title shimmer', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Title: width 80, height 22
      const titleShimmer = shimmers.find(s =>
        s.props.width === 80 && s.props.height === 22
      );

      expect(titleShimmer).toBeTruthy();
    });

    it('renders divider element', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton />);

      const divider = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          height: 1,
          backgroundColor: tokens.colors.najdi.container + '20'
        })
      });

      expect(divider).toBeTruthy();
    });

    it('renders family row container', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton />);

      const familyRow = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          flexDirection: 'row',
          gap: 12
        })
      });

      expect(familyRow).toBeTruthy();
    });
  });

  describe('Tile Structure', () => {
    it('renders avatar shimmer for each tile', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton tileCount={4} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Avatar: 40x40, borderRadius 20 (circular)
      const avatars = shimmers.filter(s =>
        s.props.width === 40 &&
        s.props.height === 40 &&
        s.props.borderRadius === 20
      );

      expect(avatars).toHaveLength(4);
    });

    it('renders name shimmer for each tile', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton tileCount={4} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Name: width 70, height 16
      const names = shimmers.filter(s =>
        s.props.width === 70 && s.props.height === 16
      );

      expect(names).toHaveLength(4);
    });

    it('renders optional label shimmer (alternating)', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton tileCount={6} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Label: width 50, height 14 (only for index % 3 === 0)
      const labels = shimmers.filter(s =>
        s.props.width === 50 && s.props.height === 14
      );

      // Tiles at index 0 and 3 should have labels (2 labels)
      expect(labels).toHaveLength(2);
    });

    it('alternates labels correctly for 9 tiles', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton tileCount={9} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Labels at index 0, 3, 6 (3 labels)
      const labels = shimmers.filter(s =>
        s.props.width === 50 && s.props.height === 14
      );

      expect(labels).toHaveLength(3);
    });
  });

  describe('Design System Compliance', () => {
    it('uses Najdi background color', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton />);

      const container = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          backgroundColor: tokens.colors.najdi.background
        })
      });

      expect(container).toBeTruthy();
    });

    it('uses correct border radius (md)', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton />);

      const container = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          borderRadius: tokens.radii.md
        })
      });

      expect(container).toBeTruthy();
    });

    it('uses correct padding (tokens.spacing)', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton />);

      const container = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          paddingHorizontal: tokens.spacing.md,
          paddingTop: tokens.spacing.md,
          paddingBottom: tokens.spacing.lg
        })
      });

      expect(container).toBeTruthy();
    });

    it('applies iOS shadow', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton />);

      const container = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          shadowColor: tokens.shadow.ios.shadowColor,
          shadowOpacity: tokens.shadow.ios.shadowOpacity,
          shadowRadius: tokens.shadow.ios.shadowRadius
        })
      });

      expect(container).toBeTruthy();
    });

    it('uses Najdi border color (40% opacity)', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton />);

      const container = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          borderWidth: 1,
          borderColor: tokens.colors.najdi.container + '40'
        })
      });

      expect(container).toBeTruthy();
    });
  });

  describe('Shimmer Count', () => {
    it('calculates correct shimmer count for 4 tiles', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton tileCount={4} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Title(1) + [Avatar(1) + Name(1) + Label(1 if index%3===0)] Ã— 4 tiles
      // Title(1) + Tile0(3) + Tile1(2) + Tile2(2) + Tile3(3) = 11
      expect(shimmers).toHaveLength(11);
    });

    it('calculates correct shimmer count for 3 tiles', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton tileCount={3} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Title(1) + Tile0(3) + Tile1(2) + Tile2(2) = 8
      expect(shimmers).toHaveLength(8);
    });

    it('calculates correct shimmer count for 1 tile', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton tileCount={1} />);
      const shimmers = UNSAFE_root.findAllByType('Shimmer');

      // Title(1) + Tile0(Avatar + Name + Label) = 4
      expect(shimmers).toHaveLength(4);
    });
  });

  describe('Edge Cases', () => {
    it('handles tileCount=0', () => {
      const { toJSON } = render(<FamilyCardSkeleton tileCount={0} />);
      expect(toJSON()).toBeTruthy();
    });

    it('handles negative tileCount (edge case)', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton tileCount={-1} />);
      const tiles = UNSAFE_root.findAllByProps({ style: expect.objectContaining({ width: 80 }) });

      // Negative length should create empty array
      expect(tiles).toHaveLength(0);
    });

    it('handles very large tileCount (20 tiles)', () => {
      const { toJSON } = render(<FamilyCardSkeleton tileCount={20} />);
      expect(toJSON()).toBeTruthy();
    });

    it('handles undefined tileCount (uses default)', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton />);
      const tiles = UNSAFE_root.findAllByProps({ style: expect.objectContaining({ width: 80 }) });

      expect(tiles).toHaveLength(4);
    });

    it('handles null tileCount', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton tileCount={null} />);
      const tiles = UNSAFE_root.findAllByProps({ style: expect.objectContaining({ width: 80 }) });

      // null should be coerced to 0
      expect(tiles).toHaveLength(0);
    });
  });

  describe('Performance', () => {
    it('renders within performance budget (< 16ms)', () => {
      const start = performance.now();
      render(<FamilyCardSkeleton />);
      const end = performance.now();

      const renderTime = end - start;
      expect(renderTime).toBeLessThan(16);
    });

    it('renders 10 tiles efficiently', () => {
      const start = performance.now();
      render(<FamilyCardSkeleton tileCount={10} />);
      const end = performance.now();

      // 10 tiles should still render quickly
      expect(end - start).toBeLessThan(30);
    });
  });

  describe('Responsive Layout', () => {
    it('maintains horizontal layout', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton />);

      const familyRow = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          flexDirection: 'row'
        })
      });

      expect(familyRow).toBeTruthy();
    });

    it('uses gap for spacing', () => {
      const { UNSAFE_root } = render(<FamilyCardSkeleton />);

      const familyRow = UNSAFE_root.findByProps({
        style: expect.objectContaining({
          gap: 12
        })
      });

      expect(familyRow).toBeTruthy();
    });
  });

  describe('Snapshot Tests', () => {
    it('matches snapshot with default props', () => {
      const { toJSON } = render(<FamilyCardSkeleton />);
      expect(toJSON()).toMatchSnapshot();
    });

    it('matches snapshot with 6 tiles', () => {
      const { toJSON } = render(<FamilyCardSkeleton tileCount={6} />);
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
