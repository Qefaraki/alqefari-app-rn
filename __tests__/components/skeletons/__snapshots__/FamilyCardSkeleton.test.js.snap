// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FamilyCardSkeleton Component Snapshot Tests matches snapshot with 6 tiles 1`] = `
<View
  style={
    {
      "backgroundColor": "#F9F7F3",
      "borderColor": "#D1BBA340",
      "borderRadius": 12,
      "borderWidth": 1,
      "marginBottom": 16,
      "paddingBottom": 20,
      "paddingHorizontal": 16,
      "paddingTop": 16,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 8,
        "width": 0,
      },
      "shadowOpacity": 0.07,
      "shadowRadius": 12,
    }
  }
>
  <Shimmer
    borderRadius={10}
    height={22}
    style={
      {
        "marginBottom": 12,
      }
    }
    width={80}
  />
  <View
    style={
      {
        "backgroundColor": "#D1BBA320",
        "height": 1,
        "marginVertical": 12,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 12,
        "paddingHorizontal": 2,
        "paddingVertical": 4,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "gap": 8,
          "width": 80,
        }
      }
    >
      <Shimmer
        borderRadius={20}
        height={40}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={40}
      />
      <Shimmer
        borderRadius={10}
        height={16}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={70}
      />
      <Shimmer
        borderRadius={10}
        height={14}
        width={50}
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "gap": 8,
          "width": 80,
        }
      }
    >
      <Shimmer
        borderRadius={20}
        height={40}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={40}
      />
      <Shimmer
        borderRadius={10}
        height={16}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={70}
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "gap": 8,
          "width": 80,
        }
      }
    >
      <Shimmer
        borderRadius={20}
        height={40}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={40}
      />
      <Shimmer
        borderRadius={10}
        height={16}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={70}
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "gap": 8,
          "width": 80,
        }
      }
    >
      <Shimmer
        borderRadius={20}
        height={40}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={40}
      />
      <Shimmer
        borderRadius={10}
        height={16}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={70}
      />
      <Shimmer
        borderRadius={10}
        height={14}
        width={50}
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "gap": 8,
          "width": 80,
        }
      }
    >
      <Shimmer
        borderRadius={20}
        height={40}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={40}
      />
      <Shimmer
        borderRadius={10}
        height={16}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={70}
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "gap": 8,
          "width": 80,
        }
      }
    >
      <Shimmer
        borderRadius={20}
        height={40}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={40}
      />
      <Shimmer
        borderRadius={10}
        height={16}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={70}
      />
    </View>
  </View>
</View>
`;

exports[`FamilyCardSkeleton Component Snapshot Tests matches snapshot with default props 1`] = `
<View
  style={
    {
      "backgroundColor": "#F9F7F3",
      "borderColor": "#D1BBA340",
      "borderRadius": 12,
      "borderWidth": 1,
      "marginBottom": 16,
      "paddingBottom": 20,
      "paddingHorizontal": 16,
      "paddingTop": 16,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 8,
        "width": 0,
      },
      "shadowOpacity": 0.07,
      "shadowRadius": 12,
    }
  }
>
  <Shimmer
    borderRadius={10}
    height={22}
    style={
      {
        "marginBottom": 12,
      }
    }
    width={80}
  />
  <View
    style={
      {
        "backgroundColor": "#D1BBA320",
        "height": 1,
        "marginVertical": 12,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 12,
        "paddingHorizontal": 2,
        "paddingVertical": 4,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "gap": 8,
          "width": 80,
        }
      }
    >
      <Shimmer
        borderRadius={20}
        height={40}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={40}
      />
      <Shimmer
        borderRadius={10}
        height={16}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={70}
      />
      <Shimmer
        borderRadius={10}
        height={14}
        width={50}
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "gap": 8,
          "width": 80,
        }
      }
    >
      <Shimmer
        borderRadius={20}
        height={40}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={40}
      />
      <Shimmer
        borderRadius={10}
        height={16}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={70}
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "gap": 8,
          "width": 80,
        }
      }
    >
      <Shimmer
        borderRadius={20}
        height={40}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={40}
      />
      <Shimmer
        borderRadius={10}
        height={16}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={70}
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "gap": 8,
          "width": 80,
        }
      }
    >
      <Shimmer
        borderRadius={20}
        height={40}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={40}
      />
      <Shimmer
        borderRadius={10}
        height={16}
        style={
          {
            "marginBottom": 4,
          }
        }
        width={70}
      />
      <Shimmer
        borderRadius={10}
        height={14}
        width={50}
      />
    </View>
  </View>
</View>
`;
